mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -355,9 +364,12 @@ object Word2VecModel extends MLReadable[Word2VecModel] {
       // Calculate the approximate size of the model.
       // Assuming an average word size of 15 bytes, the formula is:
       // (floatSize * vectorSize + 15) * numWords
-      val numWords = instance.wordVectors.wordIndex.size
-      val approximateSizeInBytes = (floatSize * instance.getVectorSize + averageWordSize) * numWords
-      ((approximateSizeInBytes / bufferSizeInBytes) + 1).toInt
+      val approximateSizeInBytes = (floatSize * vectorSize + averageWordSize) * numWords
+      val numPartitions = (approximateSizeInBytes / bufferSizeInBytes) + 1
+      require(numPartitions < 10e8, s"Word2VecModel calculated that it needs $numPartitions " +
```

Comment:
Is this failure truly necessary? Can we make this a WARN and use a best-attempt (``Int.MAX``?) partitions instead? The models that would fail here would be so huge that they likely took days to train